### PR TITLE
[KYUUBI #2063] Fix engine idle timeout lose efficacy for Flink Engine

### PR DESCRIPTION
**Pull request by yanghua**
### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Review by yaooqinn — APPROVED** [review]

File: externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/session/FlinkSQLSessionManager.scala [hunk 1]
```diff
@@ -76,6 +76,7 @@ class FlinkSQLSessionManager(engineContext: DefaultContext)
   }
 
   override def closeSession(sessionHandle: SessionHandle): Unit = {
+    super.closeSession(sessionHandle)
     executor.closeSession(sessionHandle.toString)
```
[the remark this answers is not in the document]
Comment (yanghua) [reply externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/session/FlinkSQLSessionManager.scala:80]: > Should this add the check whether the ENGINE_SHARE_LEVEL config option is equal to ShareLevel.CONNECTION to stop the session that stops the current engine?

I know this issue, Flink did not consider more about the engine share level. So I would prefer to rethink it in another thread. WDYT?

> What's more, could you please add the test case to verify whether the bug is fixed?

I have verified in my local. Writing test cases to test this issue is hard, we need to start kyuubi server, start flink local cluster then open a session and close it, and judge the engine process exited via fetching a specific log message.

We may build a better test suite for the integration test in the future. WDYT?

[the remark this answers is not in the document]
Comment (SteNicholas) [reply externals/kyuubi-flink-sql-engine/src/main/scala/org/apache/kyuubi/engine/flink/session/FlinkSQLSessionManager.scala:80]: > I know this issue, Flink did not consider more about the engine share level. So I would prefer to rethink it in another thread. WDYT?

If we rethink this issue in another discussion thread, please create a new issue for this.

> I have verified in my local. Writing test cases to test this issue is hard, we need to start kyuubi server, start flink local cluster then open a session and close it, and judge the engine process exited via fetching a specific log message.
We may build a better test suite for the integration test in the future. WDYT?

+1,the above test suite in the integration test could be together with the test suite for killing application.

**Review by SteNicholas — APPROVED** [review]
LGTM. @yaooqinn  could you please merge this bug fix to master? @yanghua , please help to create issues above mentioned.

**Comment by yaooqinn** [comment]
merged to master/1.5, thanks

**Pull request closed without merging**